apis/python/src/tiledbsoma/reindexer.cc

```diff
@@ -41,6 +41,95 @@ namespace py = pybind11;
 using namespace py::literals;
 using namespace tiledbsoma;
 
+/***
+ * Handle general lookup for Re-indexer
+ * @param indexer reference to the indexer
+ * @param lookups input values to be looked up
+ * @return looked up values
+ */
+py::array_t<int64_t> get_indexer_general(
+    IntIndexer& indexer, py::array_t<int64_t> lookups) {
+    auto input_buffer = lookups.request();
+    int64_t* input_ptr = static_cast<int64_t*>(input_buffer.ptr);
+    size_t size = input_buffer.shape[0];
+    auto results = py::array_t<int64_t>(size);
+    auto results_buffer = results.request();
+    size_t results_size = results_buffer.shape[0];
+    int64_t* results_ptr = static_cast<int64_t*>(results_buffer.ptr);
+    indexer.lookup(input_ptr, results_ptr, size);
+    return results;
+}
+
+/***
+ * Helper function to provide data and schema for an arrow object
+ * @param object python object
+ * @param arrow_array extracted array data
+ * @param arrow_schema extracted array schema
+ */
+//
+void extract_py_array_schema(
+    const pybind11::handle object,
+    ArrowArray& arrow_array,
+    ArrowSchema& arrow_schema) {
+    uintptr_t arrow_schema_ptr = (uintptr_t)(&arrow_schema);
+    uintptr_t arrow_array_ptr = (uintptr_t)(&arrow_array);
+
+    // Using array._export_to_c to get arrow array and schema.
+    object.attr("_export_to_c")(arrow_array_ptr, arrow_schema_ptr);
+}
+
+/***
+ * Handle pyarrow-based lookup for Re-indexer
+ * @param indexer reference to the indexer
+ * @py_arrow_array pyarrow inputs to be looked up
+ * @return looked up values
+ */
+py::array_t<int64_t> get_indexer_py_arrow(
+    IntIndexer& indexer, py::object py_arrow_array) {
+    // Check if it is not a pyarrow array or pyarrow chunked array
+    if (!py::hasattr(py_arrow_array, "_export_to_c") &&
+        !py::hasattr(py_arrow_array, "chunks") &&
+        !py::hasattr(py_arrow_array, "combine_chunks")) {
+        // Handle the general case (no py arrow objects)
+        return get_indexer_general(indexer, py_arrow_array);
+    }
```

Comment:
I know I said something different over DMs a few days ago, but I think it's better to remove this check from here and do these on the Python side.

Reply:
Doing it on python is not practical as the API call goes directly to the indexer meaning there's no python layer between the python `get_index` and this.

Reply:
I will create one PR with minor changes needed for this and other PRs.